sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java

```diff
@@ -223,12 +209,18 @@ public OfferEvaluator(
     private Optional<Protos.ExecutorInfo> getExecutorInfo(Collection<Protos.TaskInfo> taskInfos) {
         for (Protos.TaskInfo taskInfo : taskInfos) {
             if (taskHasReusableExecutor(taskInfo)) {
-                logger.info("Using existing executor: {}", taskInfo.getExecutor().getExecutorId().getValue());
+                logger.info("Using existing executor: {}", TextFormat.shortDebugString(taskInfo.getExecutor()));
                 return Optional.of(taskInfo.getExecutor());
             }
         }
 
-        return Optional.empty();
+        Protos.ExecutorInfo executorInfo = taskInfos.stream().findFirst().get()
+                .getExecutor().toBuilder()
+                .setExecutorId(Protos.ExecutorID.newBuilder().setValue(""))
```

Comment:
Oh I see this now. Should the field be cleared (`clearExecutorId()`) instead of set to an empty string?

Reply:
I wondered myself.  Was scared and went with empty string.  I can try it at least.

Reply:
Oh, it might throw at the `build()` call below if the field is `required`. Given that we may be stuck with an empty string.

https://github.com/apache/mesos/blob/master/include/mesos/mesos.proto#L744